packages/plugin-typescript/sources/typescriptUtils.ts

```diff
@@ -0,0 +1,31 @@
+import {Configuration, Descriptor} from '@yarnpkg/core';
+import {httpUtils, structUtils}    from '@yarnpkg/core';
+
+// Note that the appId and appKey are specific to Yarn's website - please
+// don't use them anywhere else without asking Algolia's permission
+const ALGOLIA_API_KEY = 'f54e21fa3a2a0160595bb058179bfb1e';
+const ALGOLIA_APP_ID = 'OFCNCOG2CU';
+const ALGOLIA_ENDPOINT = `https://${ALGOLIA_APP_ID}-dsn.algolia.net/1/indexes/npm-search`;
```

Comment:
Just fetching this endpoint isn’t fault-tolerant, it could fail when you need to retry. Please use the official algoliasearch package (v4 is now in alpha, and has a lot of improvements to try out here)

Reply:
@Haroenv I'm the one who suggested to use `httpUtils` - I would prefer to make all http requests go through our request layer (otherwise it's harder to keep track of the places where things like proxy need to be configured - something that `httpUtils` does for free).

Reply:
You can configure a custom requestMethod (requester: request in construction) for that reason

Reply:
Ah! That would be an option then - is it documented? I don't find it on the website, but maybe it's v4 only?

Reply:
It's a v4 option, so not yet documented. The request method on a "requester" simply needs to return a promise

Reply:
cc @nunomaduro

Reply:
Alright, if you are using the `v4`, here is how you can pass your own implementation of the requester:

```ts
algoliasearch('appId', 'apiKey', {
  requester: {
    send(request) {
      console.log(request);

      return Promise.resolve({
        content: '{ hits: [] }',
        status: 200,
        isTimedOut: false,
      });
    },
  },
});
```

Here is an example of the requester implementation: [createBrowserXhrRequester.ts](https://github.com/algolia/algoliasearch-client-javascript/blob/release/v4.0.0/packages/requester-browser-xhr/src/createBrowserXhrRequester.ts)